netcdftime/_netcdftime.pyx

```diff
@@ -1151,8 +1156,470 @@ def time2index(times, nctime, calendar=None, select='exact'):
     return _toscalar(index)
 
 
-def _toscalar(a):
+cdef _toscalar(a):
     if a.shape in [(), (1,)]:
         return a.item()
     else:
         return a
+
+cdef to_tuple(dt):
```

Comment:
The type is *not* specified here to make it possible to compare `netcdftime.datetime` to `datetime.datetime` by converting both of them to tuples. I could write a separate version exclusively for `netcdftime.datetime`, though... (Possibly implemented as a method of `netcdftime.datetime`.)